istio/src/main/scala/io/buoyant/k8s/istio/RouteCache.scala

```diff
@@ -10,7 +10,12 @@ import io.buoyant.k8s.istio.ApiserverClient.RouteRuleConfig
 import io.buoyant.namer.RichActivity
 import istio.proxy.v1.config.RouteRule
 
-class RouteCache(api: ApiserverClient) extends Closable {
+trait RouteCache extends Closable {
+  def getRules: Future[Map[String, RouteRule]]
+  val routeRules: Activity[Map[String, RouteRule]]
+}
+
+class RouteCacheBackedByApi(api: ApiserverClient) extends RouteCache {
```

Comment:
Name feels a bit verbose.

Reply:
Waiting for resolution on similar comment